interfaces/builtin/raw_volume_test.go

```diff
@@ -0,0 +1,333 @@
+	testPlugPart3Info *snap.PlugInfo
+}
+
+var _ = Suite(&rawVolumeInterfaceSuite{
+	iface: builtin.MustInterface("raw-volume"),
+})
+
+const rawVolumeConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [raw-volume]
+`
+
+const rawVolumeCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  raw-volume:
+`
+
+func (s *rawVolumeInterfaceSuite) SetUpTest(c *C) {
+	// Mock for OS snap
+	osSnapInfo := snaptest.MockInfo(c, `
+name: core
+version: 0
+type: os
+slots:
+  test-part-1:
+    interface: raw-volume
+    path: /dev/vda1
+  test-part-2:
+    interface: raw-volume
+    path: /dev/mmcblk0p1
+  test-part-3:
+    interface: raw-volume
+    path: /dev/i2o/hda1
+`, nil)
+	s.testSlot1Info = osSnapInfo.Slots["test-part-1"]
+	s.testSlot1Info = osSnapInfo.Slots["test-part-2"]
+	s.testSlot1Info = osSnapInfo.Slots["test-part-3"]
+
+	// Mock for Gadget snap
+	gadgetSnapInfo := snaptest.MockInfo(c, `
+name: some-device
+version: 0
+type: gadget
+slots:
+  test-udev-1:
+    interface: raw-volume
+    path: /dev/vda1
+  test-udev-2:
+    interface: raw-volume
+    path: /dev/mmcblk0p1
+  test-udev-3:
+    interface: raw-volume
+    path: /dev/i2o/hda1
+  test-udev-bad-value-1:
+    interface: raw-volume
+    path: /dev/vda0
+`, nil)
+	s.testUDev1Info = gadgetSnapInfo.Slots["test-udev-1"]
+	s.testUDev1 = interfaces.NewConnectedSlot(s.testUDev1Info, nil, nil)
+	s.testUDev2Info = gadgetSnapInfo.Slots["test-udev-2"]
+	s.testUDev2 = interfaces.NewConnectedSlot(s.testUDev2Info, nil, nil)
+	s.testUDev3Info = gadgetSnapInfo.Slots["test-udev-3"]
+	s.testUDev3 = interfaces.NewConnectedSlot(s.testUDev3Info, nil, nil)
+	s.testUDevBadValue1Info = gadgetSnapInfo.Slots["test-udev-bad-value-1"]
+	s.testUDevBadValue1 = interfaces.NewConnectedSlot(s.testUDevBadValue1Info, nil, nil)
+
+	// Mock for consumer snaps
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: client-snap
+version: 0
+plugs:
+  plug-for-part-1:
+    interface: raw-volume
+  plug-for-part-2:
+    interface: raw-volume
+  plug-for-part-3:
+    interface: raw-volume
+apps:
+  app-accessing-1-part:
+    command: foo
+    plugs:
+    - plug-for-part-1
+  app-accessing-2-part:
+    command: foo
+    plugs:
+    - plug-for-part-2
+  app-accessing-3-part:
+    command: foo
+    plugs:
+    - plug-for-part-3
+`, nil)
+	s.testPlugPart1Info = consumingSnapInfo.Plugs["plug-for-part-1"]
+	s.testPlugPart1 = interfaces.NewConnectedPlug(s.testPlugPart1Info, nil, nil)
+	s.testPlugPart2Info = consumingSnapInfo.Plugs["plug-for-part-2"]
+	s.testPlugPart2 = interfaces.NewConnectedPlug(s.testPlugPart2Info, nil, nil)
+	s.testPlugPart3Info = consumingSnapInfo.Plugs["plug-for-part-3"]
+	s.testPlugPart3 = interfaces.NewConnectedPlug(s.testPlugPart3Info, nil, nil)
+}
+
+func (s *rawVolumeInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "raw-volume")
+}
+
+func (s *rawVolumeInterfaceSuite) TestSanitizeCoreSnapSlot(c *C) {
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, s.testSlot1Info), IsNil)
```

Comment:
from above, should this be expanded to test the different variants declared in the yaml with multiple `testSlotXXXInfo` members on the struct?

Reply:
Done